internal/controller/kustomization_controller.go

```diff
@@ -982,15 +984,79 @@ func (r *KustomizationReconciler) checkHealth(ctx context.Context,
 		return fmt.Errorf("unable to update the healthy status to progressing: %w", err)
 	}
 
+	// Check if we should cancel health checks on new revisions
+	cancelOnNewRevision := false
+	if enabled, err := features.Enabled(features.CancelHealthCheckOnNewRevision); err == nil && enabled {
+		cancelOnNewRevision = true
+	}
```

Comment:
Please move this main.go along with all the other feature gates and pass the cancelOnNewRevision in the controller args.